File: SzachyPython/Krol.py
import Pozycja
from Figura import Figura

class Krol(Figura):

    def __init__(self, pole, kolor):
        Figura.__init__(self, pole, kolor)
        self.name = "Krol"



    def wypiszPole(self):
        return self.pozycja.wypiszPole()

    def czyDozwolonyDlaFigury(self, x, y):
        #1
        if(x == self.pozycja.x - 1 and y == self.pozycja.y+1):
            return True
        #2
        if (x == self.pozycja.x and y == self.pozycja.y + 1):
            return True
        #3
        if (x == self.pozycja.x + 1 and y == self.pozycja.y + 1):
            return True
        #4
        if (x == self.pozycja.x + 1 and y == self.pozycja.y):
            return True
        #5
        if (x == self.pozycja.x + 1 and y == self.pozycja.y - 1):
            return True
        #6
        if (x == self.pozycja.x and y == self.pozycja.y - 1):
            return True
        #7
        if (x == self.pozycja.x - 1 and y == self.pozycja.y - 1):
            return True
        #8
        if (x == self.pozycja.x - 1 and y == self.pozycja.y):
            return True

        return False


    def toString(self):
        return self.name + " " + self.kolor + " " + self.pozycja.zwrocPole()

File: szachyjava/src/Krol.java

public class Krol extends Figura{

    public String nazwa = "Krol";
    public String kolor;

    Krol(String pole, String kolorFigury) {
        super(pole);
        pozycja = new Pozycja(pole);
        this.kolor = kolorFigury;

    }

    @Override
    public void ruch(String nowePole){
        int noweX = nowePole.charAt(0) - 'A' + 1;
        int noweY = nowePole.charAt(1) - '1' + 1;

        if(czyDozwolonyRuch(noweX, noweY))
            pozycja.ustawXY(nowePole);
        else
            System.out.println("Błędny Ruch");


    }

    @Override
    public boolean czyDozwolonyDlaFigury(int noweX, int noweY, int x, int y) {

        //1
        if(noweX == x-1 && noweY == y+1)
            return true;
        //2
        if(noweX == x && noweY == y+1)
            return true;
        //3
        if(noweX == x+1 && noweY == y+1)
            return true;
        //4
        if(noweX == x+1 && noweY == y)
            return true;
        //5
        if(noweX == x+1 && noweY == y-1)
            return true;
        //6
        if(noweX == x && noweY == y-1)
            return true;
        //7
        if(noweX == x-1 && noweY == y-1)
            return true;
        //8
        if(noweX == x-1 && noweY == y)
            return true;

        return false;
    }

    public boolean czyDozwolonyRuch(int noweX, int noweY){
        //czy mieści się na planszy
        if(czyPoprawnePole(noweX, noweY) == false)
            return false;

        //czy dozwolony dla figury
        if(czyDozwolonyDlaFigury(noweX, noweY, pozycja.dajX(), pozycja.dajY()) == false)
            return false;

        return true;
    }

    @Override
    public String toString() {
        return nazwa + " " + kolor + " " + pozycja.dajPozycje();
    }
}
